File: app/src/main/java/fr/free/nrw/commons/campaigns/CampaignResponseDTO.java
package fr.free.nrw.commons.campaigns;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Data class to hold the response from the campaigns api
 */
public class CampaignResponseDTO {

    @SerializedName("config")
    private CampaignConfig campaignConfig;

    @SerializedName("campaigns")
    private List<Campaign> campaigns;

    public CampaignConfig getCampaignConfig() {
        return campaignConfig;
    }

    public List<Campaign> getCampaigns() {
        return campaigns;
    }
}

File: app/src/main/java/fr/free/nrw/commons/campaigns/CampaignResponseDTO.kt
package fr.free.nrw.commons.campaigns

import com.google.gson.annotations.SerializedName
import fr.free.nrw.commons.campaigns.models.Campaign

/**
 * Data class to hold the response from the campaigns api
 */
class CampaignResponseDTO {
    @SerializedName("config")
    val campaignConfig: CampaignConfig? = null
    @SerializedName("campaigns")
    val campaigns: List<Campaign>? = null

}